secret_santa.exs

```diff
@@ -0,0 +1,35 @@
+# The code is largely inspired by this tweet: https://twitter.com/gausby/status/1200462062733987841
+
+defmodule EnumUtils do
+  def rotate_left([h | t]) do
+    t ++ [h]
+  end
+end
+
+defmodule SecretSanta do
+  def pair([]) do
+    {:error, "No participants, no secret santas!"}
+  end
+
+  def pair([_]) do
+    {:error, "Alone for Christmas day? Watch 'Home Alone' 1 and 2!"}
+  end
+
+  def pair([_ | _] = participants) do
+    {:ok,
+     participants
+     |> Enum.shuffle()
+     |> EnumUtils.rotate_left()
```

Comment:
What's the purpose of this ?

Reply:
This are my rules of the game:
- You can't give yourself a gift
- The sequence must form a loop
- The sequence must be random

An algorithm that respects these constraints could be written like this:
1. Given a list I create a new version of the list that has been shuffled
2. Given the shuffled list I create a new version of the list that has been rotated from 1 to the left or the right
3. Zip the two list together

Pseudo code:
```
list = ["Malian", "Jehan", "Colin"]
shuffled = Shuffle(list)
rotated = Rotate(shuffled)
output = Zip(shuffled, rotated)
```

Example:
```
list = ["Malian", "Jehan", "Colin"]
shuffled = ["Malian", "Colin", "Jehan"]
rotated = ["Colin", "Jehan", "Malian"] or ["Jehan", "Malian", "Colin"]

zip(shuffled, rotated) = [
  ["Malian", "Colin"],
  ["Colin", "Jehan"],
  ["Jehan", "Malian"]
]
```

The step 2 ensures that you will not give a gift to yourself + the loop sequence.

Reply:
The python code was written for this article by Maryse. I tried to keep the same constraints for this version. https://medium.com/data-science-be/la-cacahu%C3%A8te-de-no%C3%ABl-avec-morceau-de-code-offert-37ba74943a95

Reply:
I still don't understand how rotating left helps in any way :-P. I asked the question because I was pretty sure that your code does not guarantee you don't give to yourself. 

Just to make sure, I wrote the code to test that:
```elixir
# My list of people
people = 1..20 |> Enum.into([])

# let's try this a few times
1..1000
|> Enum.map(fn _attempt ->
  people
  |> SecretSanta.pair()
  |> elem(1)
  |> Enum.any?(fn {a, b} -> a == b end)
end)
|> Enum.filter(& &1)
|> Enum.count()
|> IO.inspect(label: "attempts where a person had to give to themselves: ")
```

result: there's a collision ~2/3 of the attempts.
```bash
attempts where a person had to give to themselves: : 615
```

Reply:
Yep ! It was the case with the previous version of the code because I didn't zip with the shuffled list but with the orignal list. No, it is not the case anymore. 

I ran your test script and this is my result:
![image](https://user-images.githubusercontent.com/553321/69912447-b4ef0780-1429-11ea-8d1f-63bbd145e050.png)

Reply:
:+1: